### PR TITLE
fix(notification): make List ordering deterministic for equal timestamps

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

`InMemoryStore.List` gathered matching notifications by ranging a Go map (randomized iteration order) and then sorted them with an unstable, timestamp-only `sort.Slice`. Notifications created within the same clock tick share a `Timestamp`, which is routine on coarse-resolution clocks (Windows' ~15ms monotonic tick) and possible anywhere under load. Their relative order in `List` output, and which one a `Limit:1` query returned, was therefore arbitrary and varied from run to run.

This came up while stabilizing native Windows unit tests, where a `List(Limit:1)` returned a prior subtest's notification instead of the just-created one. The test was isolated at the time; this fixes the underlying production fragility.

## Changes

- Add a process-local monotonic creation sequence (`seq`) to `Notification`, assigned by `NewNotification` via an atomic counter and copied by `Clone`. It is unexported and never serialized or persisted (transient, like `DeliveryTarget`).
- `sortNotificationsByTime` now imposes a deterministic total order: `Timestamp` desc, then `seq` desc (most recently created wins on a tie), then `ID` asc as a final fallback for notifications built without `NewNotification` (`seq` 0). Switched from `sort.Slice` to `slices.SortFunc` + `cmp.Or`.
- `removeOldest` gains the same `seq` tiebreaker so eviction at capacity is deterministic and consistent with `List`'s ordering.

No API or wire-format change: `seq` is unexported with no JSON tag, so it never appears in REST responses. Ordering for distinct timestamps is unchanged (still newest-first).

## Test Plan

- [x] `go test -race ./internal/notification/...` passes (including `-count=5` on the new tests)
- [x] New tests cover all three comparator keys: equal-timestamp creation order (+ stable across 20 repeated calls + `Limit:1`), distinct-timestamp newest-first, and the `seq`-0 ID fallback, plus deterministic eviction
- [x] `golangci-lint run ./internal/notification/...` clean
- [x] Downstream `internal/api/v2` notification tests pass

## Summary by CodeRabbit

* **Tests**
  * Expanded coverage to verify notification listing order and capacity-based eviction are deterministic, including equal-timestamp and edge-case scenarios.
* **Bug Fixes**
  * Improved notification ordering and eviction tie-breaking to ensure consistent “newest first” behavior when multiple items share the same time.
* **Refactor**
  * Updated the internal notification ordering logic to use a fully deterministic strategy so list results and eviction decisions remain aligned.

### CODE REVIEW

File: internal/notification/types.go [hunk 1]
```diff
@@ -558,16 +578,27 @@ func (s *InMemoryStore) DeleteExpired() error {
...
-		if oldestID == "" || notif.Timestamp.Before(oldestTime) {
-			oldestID = id
-			oldestTime = notif.Timestamp
+		switch {
+		case oldestID == "": // first candidate
+		case notif.Timestamp.Before(oldestTime): // strictly older
+		case notif.Timestamp.Equal(oldestTime) && notif.seq < oldestSeq: // same time, created earlier
+		default:
+			continue
 		}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/notification/types.go]: Good catch, added the ID fallback in 916f67b8. removeOldest now mirrors sortNotificationsByTime inverted (earliest Timestamp, then lowest seq, then highest ID), so the seq-tie / seq-0 case is deterministic and consistent with List ranking the highest ID last.

File: internal/notification/types.go [hunk 2]
```diff
@@ -558,16 +578,27 @@ func (s *InMemoryStore) DeleteExpired() error {
...
+		oldestID, oldestTime, oldestSeq = id, notif.Timestamp, notif.seq
 	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/notification/types.go:609]: Fixed in 916f67b8. One correction on the suggested direction though: List sorts ID ascending, so it ranks the *highest* ID last, which means eviction should drop the highest ID. The snippet's `strings.Compare(id, oldestID)` is ascending (it would evict the smallest ID, i.e. the entry List ranks first). I mirrored the List comparator inverted instead: earliest Timestamp, then lowest seq, then highest ID (`strings.Compare(oldestID, id)`). Added a test for the seq-0 eviction path.

**Comment by tphakala** [comment]
@coderabbitai review

File: internal/notification/types.go [hunk 3]
```diff
@@ -4,11 +4,13 @@
...
 import (
+	"cmp"
 	"fmt"
 	"reflect"
 	"slices"
-	"sort"
+	"strings"
 	"sync"
+	"sync/atomic"
 	"time"
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/notification/types.go:13]: Done in c75f3151. Swapped both strings.Compare sites to cmp.Compare and removed the now-unused strings import, so the comparators are uniformly cmp-based.

File: internal/notification/types.go [hunk 4]
```diff
@@ -558,15 +578,34 @@ func (s *InMemoryStore) DeleteExpired() error {
...
+			oldestID, oldestTime, oldestSeq = id, notif.Timestamp, notif.seq
+			continue
+		}
+		// Mirror sortNotificationsByTime, inverted to find the element it ranks
+		// last: earlier Timestamp, then lower seq, then higher ID. The ID args
+		// are reversed (oldestID, id) so a larger id compares as "older".
+		older := cmp.Or(
+			notif.Timestamp.Compare(oldestTime),
+			cmp.Compare(notif.seq, oldestSeq),
+			strings.Compare(oldestID, id),
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/notification/types.go]: Done in c75f3151.

File: internal/notification/types.go [hunk 5]
```diff
@@ -627,10 +666,20 @@ func (s *InMemoryStore) GetUnreadCount() (int, error) {
...
+// Go map, whose iteration order is randomized; a timestamp-only comparison
+// would leave equal-timestamp notifications in arbitrary, run-varying order.
 func sortNotificationsByTime(notifications []*Notification) {
-	sort.Slice(notifications, func(i, j int) bool {
-		return notifications[i].Timestamp.After(notifications[j].Timestamp)
+	slices.SortFunc(notifications, func(a, b *Notification) int {
+		return cmp.Or(
+			b.Timestamp.Compare(a.Timestamp), // newest first
+			cmp.Compare(b.seq, a.seq),        // later creation first on equal timestamps
+			strings.Compare(a.ID, b.ID),      // stable fallback for seq-less notifications
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/notification/types.go]: Done in c75f3151.
